File: lib/models/search_condition.dart
class SearchCondition {
  final String model;
  final bool isRented;
  final double minPrice;
  final double maxPrice;
  final int parkingId;

  SearchCondition({
    required this.model,
    required this.isRented,
    required this.minPrice,
    required this.maxPrice,
    required this.parkingId,
  });

  SearchCondition copyWith({
    String? model,
    bool? isRented,
    double? minPrice,
    double? maxPrice,
    int? parkingId,
  }) {
    return SearchCondition(
      model: model ?? this.model,
      isRented: isRented ?? this.isRented,
      minPrice: minPrice ?? this.minPrice,
      maxPrice: maxPrice ?? this.maxPrice,
      parkingId: parkingId ?? this.parkingId,
    );
  }
}

File: src/main/java/org/example/DTO/SearchCondition.java
package org.example.DTO;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class SearchCondition {
    private String model;       // 모델명
    private Boolean isRented;   // 대여 중 여부 필터
    private Double minPrice;    // 최소 가격
    private Double maxPrice;    // 최대 가격
    private Integer parkingId;  // 특정 주차장 필터
}
